acronym/utils/cordis.py

```diff
@@ -0,0 +1,190 @@
+import pandas as pd
+import pathlib
+import numpy as np
+import os
+from toolz.functoolz import pipe
+from typing import List, Tuple, Union, Optional
+
+from acronym import PROJECT_DIR, get_yaml_config
+from acronym.utils.io import extractall, fetch
+from acronym.utils.text import camel_to_snake
+
+
+CORDIS_DATA_DIR = PROJECT_DIR / "inputs/data/cordis/"
+CONFIG = get_yaml_config(PROJECT_DIR / "acronym/config/cordis.yml")
+
+
+def cordis_file_path(
+    fp: str,
+    sub_dir: Optional[str] = None,
+) -> pathlib.Path:
+    """cordis_file_path
+    Create the file path for datasets for a CORDIS Framework Programme.
+
+    Args:
+        fp (str): Abbreviation of Framework Programme.
+        sub_dir (str): Name of optional subdirectory. E.g. to create a sub
+            directory for individual project XML files.
+
+    Returns:
+        (str): File path
+    """
+    if sub_dir is None:
+        return CORDIS_DATA_DIR / fp
+    else:
+        return CORDIS_DATA_DIR / f"{fp}/{sub_dir}"
```

Comment:
```suggestion
def cordis_file_path(
    fp: str,
    sub_dir: str = ""
) -> pathlib.Path:
    """cordis_file_path
    Create the file path for datasets for a CORDIS Framework Programme.

    Args:
        fp (str): Abbreviation of Framework Programme.
        sub_dir (str): Name of optional subdirectory. E.g. to create a sub
            directory for individual project XML files.

    Returns:
        (str): File path
    """
    return CORDIS_DATA_DIR / f"{fp}/{sub_dir}"
```
In the long term, this could be replaced by a thin wrapper around a more generic function that takes in *paths